ibis/pyspark/tests/test_timecontext.py

```diff
@@ -13,3 +18,213 @@ def test_table_with_timecontext(client):
     expected = table.compile().toPandas()
     expected = expected[expected.time.between(*context)]
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    ('ibis_window', 'spark_range'),
+    [
+        param(
+            ibis.trailing_window(
+                preceding=ibis.interval(hours=1),
+                order_by='time',
+                group_by='key',
+            ),
+            (-3600, 0),
+        ),  # 1h back looking window
+        param(
+            ibis.trailing_window(
+                preceding=ibis.interval(hours=2),
+                order_by='time',
+                group_by='key',
+            ),
+            (-7200, 0),
+        ),  # 2h back looking window
+        param(
+            ibis.range_window(
+                preceding=0,
+                following=ibis.interval(hours=1),
+                order_by='time',
+                group_by='key',
+            ),
+            (0, 3600),
+        ),  # 1h forward looking window
+    ],
+)
+def test_window_with_timecontext(client, ibis_window, spark_range):
+    """ Test context adjustment for trailing / range window
+
+    We expand context according to window sizes, for example, for a table of:
+    time       value
+    2020-01-01   a
+    2020-01-02   b
+    2020-01-03   c
+    2020-01-04   d
+    with context = (2020-01-03, 2002-01-04) trailing count for 1 day will be:
+    time       value  count
+    2020-01-03   c      2
```

Comment:
Why is this 2 and 3? Shouldn't `2020-01-03` and `2020-01-04` have the same trailing 1 day count?

Reply:
My bad in writing doc, the behavior is right.